coremltools/models/utils.py

```diff
@@ -1675,3 +1681,394 @@ def randomize_weights(mlmodel: "_ct.models.MLModel"):
     )
 
     return randomized_mlmodel
+
+
+def bisect_model(
+    model_path: str,
+    output_dir: str,
+    remove_original: _Optional[bool] = False,
+    merge_chunks_to_pipeline: _Optional[bool] = False,
+    check_output_correctness: _Optional[bool] = False,
+):
+    """
+    Utility function to split a mlpackage model into two mlpackages of approximately same file size.
+
+    The implementation of this util is mainly migrated from the below source:
+    https://github.com/apple/ml-stable-diffusion/blob/main/python_coreml_stable_diffusion/chunk_mlprogram.py
+
+    Parameters
+    ----------
+    model_path: str
+        Path to the mlpackage file to be split into two mlpackages of approximately same file size.
+
+    output_dir: str
+        Path to output directory where the two model chunks / pipeline model would be saved.
+        If the `model_path` is `{path}/{model_name}.mlpackage`, the chunks models are going to be saved as:
+
+        1. first chunk model: `{output_dir}/{model_name}_chunk1.mlpackage`
+        2. second chunk model: `{output_dir}/{model_name}_chunk2.mlpackage`
+        3. chunked pipeline model: `{output_dir}/{model_name}_chunked_pipeline.mlpackage`
+
+    remove_original: bool
```

Comment:
This option seems a little strange to me. IMO the API would be cleaner without it.

Reply:
+1, is the storage too huge to tolerate? Imho
1. User can decide whether delete or not themselves
2. Usually we don't have to save peak hard disk usage

Reply:
Agree. I think we can handle this in the ml-stable-diffusion repo itself @aseemw

Reply:
Done!